src/aspire/basis/polar_2d.py

```diff
@@ -80,6 +80,9 @@ def evaluate(self, v):
             coordinate basis. This is an array whose first two dimensions equal `self.sz`
             and the remaining dimensions correspond to dimensions two and higher of `v`.
         """
+        if self.dtype != real_type(v.dtype):
+            logger.error('Data type is not consistent with the defined in the class.')
```

Comment:
Can you specify the data types (for the class and for the input) in the message? Also below.

Reply:
Done.

Reply:
Ok thanks.